File: SportsBettingApp/src/types/WalletTransaction.ts

enum TransactionType {
  Deposit,
  Withdraw,
  Bet,
  Win
}

interface WalletTransaction
{
    id?: number;
    transactionTimestamp: string;
    transactionAmount: number;
    transactionType: TransactionType;
    walletFinalAmount: number;
}

export { WalletTransaction, TransactionType };

File: SportsBettingApp-Backend/Domain/Models/WalletTransaction.cs
global using Domain.Common;
using System.Text.Json.Serialization;

namespace Domain.Models
{
    public enum TransactionType
    {
        Deposit,
        Withdraw,
        Bet,
        Win
    }

    public class WalletTransaction : BaseEntity
    {
        public DateTime TransactionTimestamp { get; set; }

        public double TransactionAmount { get; set; }

        public TransactionType TransactionType { get; set; }

        public double WalletFinalAmount { get; set; }

    }
}
